features/cellular/framework/common/CellularUtil.cpp

```diff
@@ -378,6 +378,8 @@ pdp_type_t string_to_pdp_type(const char *pdp_type_str)
         pdp_type = IPV4_PDP_TYPE;
     } else if (len == 6 && memcmp(pdp_type_str, "Non-IP", len) == 0) {
         pdp_type = NON_IP_PDP_TYPE;
+    } else if (len == 5 && memcmp(pdp_type_str, "NONIP", len) == 0) {
```

Comment:
Is this now a bit problematic if modem has a non-standard string for Non-IP? I think this should be connected to `get_nonip_context_type_str()` method somehow.

Reply:
So far, I have seen only `NONIP` and `Non-IP` strings and this is an easy way to add support without breaking existing APIs.

Reply:
How about giving `get_nonip_context_type_str()` as a function pointer parameter to `string_to_pdp_type()` . Then as a last step it would check match from that?

Reply:
Wouldn't that break `string_to_pdp_type` API?

Reply:
That's ok for mbed-os 6.

Reply:
`string_to_pdp_type()` is only used in cellular context classes so maybe it should be moved to `CellularContext` in mbed-os 6. Then it can directly call `get_nonip_context_type_str()`.